meilisearch-release-changelog/v1.2.0.md

```diff
@@ -0,0 +1,203 @@
+
+```json
+[{
+    "id": 0,
+    "color": []
+},
+{
+    "id": 1,
+    "color": null
+},
+{
+    "id": 2,
+    "color": ""
+},
+{
+    "id": 3,
+    "color": {}
+},
+{
+    "id": 4
+}]
+```
+
+The filter `color IS EMPTY` selects the document ids `0`, `2` and `3` while the filter `color IS NOT EMPTY` or `NOT color IS EMPTY` selects the document ids `1` and `4`.
+
+The filter `color IS NULL` selects the document ids `1` while the filter `color IS NOT NULL` or `NOT color IS NULL` selects the document ids `0`, `2`, `3` and `4`.
+
+Done by @Kerollmops in #3571.
+
+## Improve relevancy and search performance
+
+v1.2 brings a massive refactor of the search engine that brings perforance and relevancy improvements. Also, one of the major benefits of this refactoring is the ability to explore and integrate new features related to the search in the future releases. Stay tuned!
+
+Most of the points linked below can be really technical. If you have any questions or unexpected behaviors on your side, feel free to [reach us](https://github.com/meilisearch/meilisearch/discussions).
+
+Done by @loiclec, @dureuill, @ManyTheFish, @Kerollmops and @irevoire in #3542.
+
+### Performance improvements
+
+- The fastest 75 percent of queries now consistently answer below 50ms on our test setup.
+- The complexity of a search query is now limited, in order to limit CPU time and RAM consumption. In more details:
+  - The number of typo considered for a single term is now limited to 150 for 1 typo, 50 for 2 typos. Similarly, the number of considered (single word or multi word) synonyms is limited to 50, the total number of words for all synonyms of a term is limited to 100.
+  - The number of words considered in a query is now limited to 1000. The number of terms (words not inside a phrase, and individual phrases) was and is still limited to 10, so this change only impacts queries containing very long phrases.
+- Geo search improvements
+  - Improved performance when there are only a few documents to geosort.
+  - Descending sort is now as performant as ascending sort.
+- Solves some performance issues: https://github.com/meilisearch/meilisearch/issues/3124, https://github.com/meilisearch/meilisearch/issues/3378, https://github.com/meilisearch/meilisearch/issues/2202, https://github.com/meilisearch/meilisearch/issues/3123
+
+### Relevancy improvements
+
+- Synonyms are no longer considered as exact matches by the `exactness` ranking rule. This change is due to the goal of the `exactness` ranking rule of boosting the documents that contain the query exactly as typed by the user. Other ranking rules will still consider synonyms. Solves https://github.com/meilisearch/meilisearch/issues/3410
+- `Attributes`, `Exactness`, `typo` and `proximity` ranking rules now imply `words` if `words` was not already specified before them. This change was made because the behavior of applying these ranking rules before or without `words` was unintuitive and could hurt relevancy.
```

Comment:
Really complexe to understand to me, outside of the squad

Reply:
Agree with @curquiza, I'm really struggling to understand this one. Specifically, what do we mean by `imply` here?

Does it mean that, if I set ranking rules to something like:

```json
[
  "typo",
  "proximity",
]
```

Meilisearch will act as if `words` was still there and at the top of the relevancy chain? E.g.:

```js
[
  // "words",  not here explicitly, but still affecting relevancy
  "typo",
  "proximity",
]
```

If my understanding is correct, I suggest we update this line to:

```suggestion
- When using the `attributes`, `exactness`, `typo`, and `proximity` ranking rules, Meilisearch will sort results as if  the`words` ranking rule were present and with higher priority. Search behavior when using the specified ranking rules before or without `words` was confusing and hurt relevancy
```

Reply:
@guimachiavelli  your understanding is correct